### DISCUSSION TITLE
cannot build helix with craneLibStable after MSRV bump

### DISCUSSION THREAD

**Discussion by evanrichter** [helix-editor/helix#8940] [Troubleshooting]
### Summary

I use this repository (or various PRs at times) as a flake input to use and test the latest helix on my system. The recent MSRV bump was from rustc 1.69.0 -> 1.70.0 in [this commit](https://github.com/helix-editor/helix/commit/8b0ae3d27912799d59e4948ff11d3a5428e32ce4), but because the rust-overlay input was not also updated, the `rust-bin.stable.latest` [is still being tracked as 1.69.0](https://github.com/oxalica/rust-overlay/blob/c57e210faf68e5d5386f18f1b17ad8365d25e4ed/manifests/stable/default.nix#L62), which breaks the nix build:

https://github.com/helix-editor/helix/blob/b023faacf898e2f393fa8f7f91e685547080442c/flake.nix#L125

Should `nix flake check` be added to CI? There are some decent caching tools now that shouldn't make it *too* slow

### Platform

nix

### Terminal Emulator

n/a

### Installation Method

nix flake

### Helix Version

b023faacf898e2f393fa8f7f91e685547080442c

**Comment by the-mikedavis** [comment]
That commit bumps the MSRV from 1.65 (not 1.69) to 1.70.

I see rust-overlay currently locked at `f335a0213504c7e6481c359dc1009be9cf34432c`: https://github.com/helix-editor/helix/blob/b023faacf898e2f393fa8f7f91e685547080442c/flake.lock#L101-L102

which has 1.71 as the latest version: https://github.com/oxalica/rust-overlay/blob/f335a0213504c7e6481c359dc1009be9cf34432c/manifests/stable/default.nix#L64

The flake builds fine for me on master with just `nix build`, installing into a profile or running `nix flake check` and we do build and cachix the packages in CI, for example https://github.com/helix-editor/helix/actions/runs/7020762475/job/19101244527.

Doing `nix flake check` in CI wouldn't help here: it's building with `craneLibMsrv` rather than `craneLibStable`.

So I don't see how the flake build is broken unless you aren't running master. Regardless I would approve a PR that updates the flake dependencies (since that updates other stuff like rust-analyzer too).

**Comment by evanrichter** [comment]
> That commit bumps the MSRV from 1.65 (not 1.69) to 1.70.

oops! I must have been thinking about the rust-overlay issue I was seeing.

What's further VERY strange, is I can't reproduce my issue earlier. In my [amp-jump branch](https://github.com/evanrichter/helix/commits/amp-jump) is Pascal's PR rebased on master so that tree-sitter-gemini could build. There were no rebase conflicts, but I definitely was getting errors saying the current rustc was 1.69.0 but helix-dap (I think?) needed 1.70.0.

And that eventually led me to check the flake.lock to see what rust-overlay version it was choosing for latest stable and I had c57e210faf68e5d5386f18f1b17ad8365d25e4ed from when helix last updated in May (and I have no other uses of rust-overlay in my system flake). I only then added the rust-overlay bump and then I could proceed. But now I can definitely build without that last version bump :upside_down_face: 

I'll just close as this seems to not be reproducible somehow, which is really not what I've come to expect from nix
